test/mocha/reserved-class.js

```diff
@@ -0,0 +1,27 @@
+var Uglify = require('../../');
+var assert = require("assert");
+
+describe("mangle: reserved_class", function () {
+    var input_js = `
+(function() {
+    class Foo extends HTMLElement {}
+    Foo.reg;
+    class Hello extends HTMLElement {} 
+    Hello.reg;
+})();   
+`;
+    var output_js = '!function(){class e extends HTMLElement{}e.reg;class Hello extends HTMLElement{}Hello.reg}();'
+
+    it("Should test mangle with reserved_class.", function () {
+        var result = Uglify.minify(input_js, {
+            mangle: { 
+                reserved: ['Hello'],
+                properties: false,
+                keep_classnames: false,
+            },
+        });
+        assert.strictEqual(result.code, output_js);
+    });
+
+
+});
```

Comment:
There is no need for a mocha test.

Please make a test case within `test/compress/harmony.js` (or in a new file say `test/compress/issue-2316.js).

Reply:
Example test to use as a guideline:

https://github.com/mishoo/UglifyJS2/blob/69cb459c16b1a264f95681e8414c4d51a3197082/test/compress/properties.js#L251-L302